### PR TITLE
Turn off crop stress

### LINKED ISSUE #1519: [CS][Pilot Testing] Make crop stress configurable by users
**Is your feature request related to a problem? Please describe.**
When calculating how much biomass a crop accumulates on a given day, the amount of stress the crop experiences from extreme temperatures, and lack of water, N, or P will reduce how much biomass is actually gained. If these stressors are really high, RuFaS will project unrealistically low crop growth.

**Describe the solution you'd like**
There should be a flag in the inputs indicating whether or not crop stressors should effect crop growth.

There are multiple possible locations where this flag could go - the config input, the input for an individual field, the crop rotation input. It is my opinion that the best place to put it would be in the input for individual fields.

**Describe alternatives you've considered**
An alternative could be allowing fully customizable crops, in which there is a configuration for whether or not the crop experiences stresses. If/when that feature gets implemented, it should be the _only_ way to configure whether or not a crop experiences stressors. But since that is a large undertaking, the solution described above should be used as a stand-in until that feature is implemented.

**Additional context**
This feature was requested by @krb78, so she should be consulted first before deciding on how this feature will be implemented.

### PR DESCRIPTION
**Pull request by Isgotkowitz**
This PR allows users to choose which stressors experience crop growth in a simulated field.

### Context

Issue(s) closed by this pull request: closes #1519

- [x] The [changelog](https://docs.google.com/spreadsheets/d/1U-3igxdstHTFb6k5dVcE5-x9IL0r6q18eUiwwytKZyA/edit#gid=0) has been updated.

### What

Inputs added to field specifications allow users to turn individual stresses on and off for all crops grown in that field. The stresses available to configure are water, temperature, nitrogen, and phosphorus related - all the ones that RuFaS currently simulates.

### Why

This allows users to be more precise and simulate only the aspects of crop growth that they want to simulate. It also makes it possible to see the examine the effect that temperature and availability of water, nitrogen, and phosphorus have on the final growth outputs.

### How

4 boolean inputs have been added to field specification files. These inputs are held in `FieldData`, and are passed to the `constrain_growth` method of the `GrowthConstraints` class when it is called in `Field`. Once in `constrain_growth`, each bool is used to determine if the corresponding stress is calculated or set to 0.

The method `grow_crop` in `Crop` has also been updated, but this method is currently unused.

All field inputs files have been updated to include inputs for the stressors. Also, the soil profiles have been updated to include the initial amounts of organic phosphorus in each soil layer. This change gets rid of some of the warnings that were being raised.

### Test plan

All unit tests associated with changed code have been updated.

To check the outputs, I tried running the same simulation but turning the stressors on and off to check that the stresses were being set correct and that final yields were changing. These outputs can be collected with the following filter:
```json
{
  "multiple": [
    {
      "name": "Yield",
      "filters": ["CropManagement._record_yield.harvest_yield.field='.*'"],
      "variables": ["crop", "wet_yield", "dry_yield", "planting_date", "harvest_date"]
    },
    {
      "name": "Stressors",
      "filters": ["FieldDataReporter.send_daily_variables\\..*stress.*"]
    }
  ]
}
```

### CODE REVIEW

**Review by krb78 — APPROVED** [review]
Looks great to me from a code perspective and better yet, it works perfectly on my stressed out crops :)

File: RUFAS/routines/field/crop/crop.py [hunk 1]
```diff
@@ -102,6 +106,14 @@ def grow_crop(
             Minimum air temperature for the day (°C).
         max_air_temperature : float
             Maximum air temperature for the day (°C).
+        simulate_water_stress : bool
+            Whether water stress should affect growth of the crop grown.
```
Comment (ew3361zh) [inline RUFAS/routines/field/crop/crop.py]:
```suggestion
            Whether water stress should affect crop growth.
```

File: RUFAS/routines/field/crop/crop.py [hunk 2]
```diff
@@ -102,6 +106,14 @@ def grow_crop(
...
+        simulate_temp_stress : bool
+            Whether temperature stress should affect growth of the crop grown.
```
Comment (ew3361zh) [inline RUFAS/routines/field/crop/crop.py]:
```suggestion
            Whether temperature stress should affect crop growth.
```

File: RUFAS/routines/field/crop/crop.py [hunk 3]
```diff
@@ -102,6 +106,14 @@ def grow_crop(
...
+        simulate_nitrogen_stress : bool
+            Whether nitrogen stress should affect growth of the crop grown.
```
Comment (ew3361zh) [inline RUFAS/routines/field/crop/crop.py]:
```suggestion
            Whether nitrogen stress should affect crop growth.
```

File: RUFAS/routines/field/crop/crop.py [hunk 4]
```diff
@@ -102,6 +106,14 @@ def grow_crop(
...
+        simulate_phosphorus_stress : bool
+            Whether phosphorus stress should affect growth of the crop grown.
```
Comment (ew3361zh) [inline RUFAS/routines/field/crop/crop.py]:
```suggestion
            Whether phosphorus stress should affect crop growth.
```

**Review by ew3361zh — APPROVED** [review]
Worked well with each of the scenarios. I also tried with turning some of the stressors off and leaving some of them on and it didn't seem to cause any issues.

A couple suggestions around wording. I don't know if the exact wording you started with is necessary but if so please feel free to ignore. It's possible I'm not getting all the nuance of the original phrasing.

**Comment by Isgotkowitz** [comment]
> Worked well with each of the scenarios. I also tried with turning some of the stressors off and leaving some of them on and it didn't seem to cause any issues.
> 
> A couple suggestions around wording. I don't know if the exact wording you started with is necessary but if so please feel free to ignore. It's possible I'm not getting all the nuance of the original phrasing.

Thank you @ew3361zh these were very good suggestions, the docstrings read way better now

**Comment by Isgotkowitz** [comment]
> Looks great to me from a code perspective and better yet, it works perfectly on my stressed out crops :)

Now I just need some flags to turn stressors on and off in my life

**Pull request merged**